File: node/test/application.test.js
/**
 * Copyright (c) 2015-2022, FineX, All Rights Reserved.
 * @author Thomas.Zhuang
 * @date 2022/05/17
 */

import { ApplicationApi } from '../src/application';

export const ApiKey = 'TDlysl39yzl65V0ZmVf6AcSJTL3VwGYp';

describe('application test', () => {
    const api = new ApplicationApi(ApiKey);
    let appId = '';

    test('appList', async () => {
        const appList = await api.appList();
        expect(appList.apps).toBeTruthy();
        console.log(appList);
        appId = appList.apps[0].app_id;
    });

    test('entryList', async () => {
        const entryList = await api.entryList(appId);
        expect(entryList.forms).toBeTruthy();
        console.log(entryList);
    });
});

File: go/application_test.go
/**
 * Copyright (c) 2015-2022, FineX, All Rights Reserved.
 * @author Thomas.Zhuang
 * @date 2022/5/17
 */

package _go

import (
	"encoding/json"
	"testing"
)

const ApiKey = "TDlysl39yzl65V0ZmVf6AcSJTL3VwGYp"
const Host = "https://api.jiandaoyun.com/api"

// JsonIndentString json换行缩进字符串
func JsonIndentString(v interface{}) string {
	indent, _ := json.MarshalIndent(v, "", "    ")
	return string(indent)
}

func TestApplicationApi(t *testing.T) {
	api := new(ApplicationApi)
	api.Api = &Api{ApiKey: ApiKey, Host: Host}

	t.Run("AppList", func(t *testing.T) {
		list, err := api.AppList()
		if err != nil {
			t.Error(err)
		} else {
			t.Log(JsonIndentString(list))
		}
	})
}
